MD5.xs: silence possible loss of data warnings on Win32

demerphq opened the pull request:
When building Perl on Win32 we see warnings like this:

    MD5.xs(723) : warning C4244: '=' : conversion from 'U32' to 'char', possible loss of data
    MD5.xs(723) : warning C4244: '=' : conversion from 'U32' to 'char', possible loss of data
    MD5.xs(723) : warning C4244: '=' : conversion from 'U32' to 'char', possible loss of data
    MD5.xs(724) : warning C4244: '=' : conversion from 'U32' to 'char', possible loss of data
    MD5.xs(724) : warning C4244: '=' : conversion from 'U32' to 'char', possible loss of data
    MD5.xs(724) : warning C4244: '=' : conversion from 'U32' to 'char', possible loss of data
    MD5.xs(725) : warning C4244: '=' : conversion from 'U32' to 'char', possible loss of data

We would like Perl to build without warnings on all platforms so we
would like to get these warnings silenced. You don't see them on gcc
builds as gcc has different ideas about what to warn for in these cases.
You can enable -Wconversion but it complains about a ton of stuff that
VC wouldn't. Generally VC warnings are at least plausibly mistakes,
the -Wconversion ones are mostly things that are not. :-(

vpit committed:
Preserve OPTIMIZE in hints

The irix_6 hints file forcefully overwrite OPTIMIZE in order to work around
a compiler mishap with high level optimizations. However, this loses
whatever extra flags are listed in OPTIMIZE, such as compiler warnings
flags or PERL_POISON, and makes the module difficult to run under a
debugger since a minimal level of optimization is always enforced. This may
also cause loading errors with the new XS handshake facility.

Instead, we chose to just follow Storable's strategy of lowering the
optimization level with a substitution, while keeping all the other flags
untouched. If other compiler flags are deemed problematic (such as
-mcpu/-march on gcc), they ought to be addressed separately.

afresh1 committed:
Add a failing test for restoring context

afresh1 committed:
Don't return context buffer if there isn't one

If we don't have a remaining buffer, don't return the garbage that might
come from passing 0 to newSVpv and letting perl try to calculate the
length of the string.

Fixes #9

demerphq committed:
MD5.xs: silence possible loss of data warnings on Win32

When building Perl on Win32 we see warnings like this:

    MD5.xs(723) : warning C4244: '=' : conversion from 'U32' to 'char', possible loss of data
    MD5.xs(723) : warning C4244: '=' : conversion from 'U32' to 'char', possible loss of data
    MD5.xs(723) : warning C4244: '=' : conversion from 'U32' to 'char', possible loss of data
    MD5.xs(724) : warning C4244: '=' : conversion from 'U32' to 'char', possible loss of data
    MD5.xs(724) : warning C4244: '=' : conversion from 'U32' to 'char', possible loss of data
    MD5.xs(724) : warning C4244: '=' : conversion from 'U32' to 'char', possible loss of data
    MD5.xs(725) : warning C4244: '=' : conversion from 'U32' to 'char', possible loss of data

We would like Perl to build without warnings on all platforms so we
would like to get these warnings silenced. You don't see them on gcc
builds as gcc has different ideas about what to warn for in these cases.
You can enable -Wconversion but it complains about a ton of stuff that
VC wouldn't. Generally VC warnings are at least plausibly mistakes,
the -Wconversion ones are mostly things that are not. :-(

afresh1 committed:
Fix test iterations

toddr committed:
Enable strict and warnings on all code.

toddr committed:
Update .gitignore/MANIFEST

toddr committed:
Enable Github actions

toddr committed:
Merge pull request #11 from Dual-Life/strict

Strict

toddr commented:
@demerphq I have moved this case to here https://github.com/gisle/digest-md5/pull/19 for merge and release.

toddr commented:
@demerphq I have moved this case here https://github.com/Dual-Life/digest-md5/pull/13 for merge and release.

toddr committed:
Merge pull request #13 from Dual-Life/yves-win32

MD5.xs: silence possible loss of data warnings on Win32

toddr committed:
Merge branch 'master' into issue_9

toddr committed:
Merge pull request #14 from Dual-Life/vincent-pit-hints

Preserve OPTIMIZE in hints

toddr committed:
Merge pull request #12 from afresh1/issue_9

Fix context save/restore when buffer is at zero

toddr committed:
Remove done_testing to make perl 5.8 CI happy

toddr committed:
Bump to 2.56 for release to CPAN

toddr committed:
Cherry pick from blead:

commit afd78941e992dbbdbeee994cb9f72f6124544080
Author: Vincent Pit <perl@profvince.com>
Date:   Tue Mar 17 15:03:24 2015 -0300

Preserve OPTIMIZE in hints

Fixes #7

The irix_6 hints file forcefully overwrite OPTIMIZE in order to work around
a compiler mishap with high level optimizations. However, this loses
whatever extra flags are listed in OPTIMIZE, such as compiler warnings
flags or PERL_POISON, and makes the module difficult to run under a
debugger since a minimal level of optimization is always enforced. This may
also cause loading errors with the new XS handshake facility.

Instead, we chose to just follow Storable's strategy of lowering the
optimization level with a substitution, while keeping all the other flags
untouched. If other compiler flags are deemed problematic (such as
-mcpu/-march on gcc), they ought to be addressed separately.

toddr committed:
Merge pull request #15 from Dual-Life/byte_order

Byte order

toddr committed:
Bump to 2.57 for release to CPAN

toddr committed:
Update repo/issues locations. Make recent changes to known standards

toddr committed:
Bump to 2.58 for release to CPAN
